Source/FieldSolver/WarpXExternalEMFields.cpp

```diff
@@ -85,19 +94,34 @@ WarpX::ApplyExternalFieldExcitationOnGrid (
                 amrex::Real x, y, z;
                 WarpXUtilAlgo::getCellCoordinates(i, j, k, mfx_stag,
                                                   problo, dx, x, y, z);
-                Fx(i, j, k) += xfield_parser(x, y, z, t);
+                if ( xflag_parser(x,y,z) == 0 ) {
+                    Fx(i, j, k) += xfield_parser(x, y, z, t);
+                }
+                if ( xflag_parser(x,y,z) == 1 ) {
+                    Fx(i, j, k) = xfield_parser(x, y, z, t);
+                }
```

Comment:
Would it look okay to change this to:
`Fx(i, j, k) = Fx(i, j, k) * xflag_parser(x,y,z) + xfield_parser(x, y, z, t);`

Reply:
@jackieyao0114 Thank you for this solution! I think it is a great idea provided the flag_parser is 0/1.
But now we have a default of -1 to force the user to specify the type of excitation.

Reply:
0 would have to mean hard source, and 1 would have to be soft source.
And you would only enter this is flag_parser(i,j,k) >= 0
I'm ok with this as long as a comment is left in the code explaining this.

Reply:
(hard is =, and soft is +=, right?)

Reply:
it's also a clarity thing... code is easier to understand the way it is
If the code is left as is, you may as well change the second "if" statement to and "else if" for a tiny speedup.

Reply:
yeah, hard is = and soft is +=. So if we use my suggested code we have to flip the flags. I'm okay either way --- keep it as is or change it

Reply:
just changed it to else if

Reply:
But I will be happy to add this one-line elegance as well. and change the definition of hard-source and soft source in the flag! :)